File: 0645-set-mismatch/0645-set-mismatch.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        D=dict(Counter(nums)) 
        ans=[0 for i in range(2)]
        for i in range(1,len(nums)+1):
            if D.get(i,0)==0:
                ans[1]=i
            if D.get(i,0)==2:
                ans[0]=i
        return ans
                

File: 0645-set-mismatch/0645-set-mismatch.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        vector<int>arr(nums.size(),0);
        for(auto x:nums)arr[x-1]++;
        vector<int>ans(2,0);
        for(int x=0;x<arr.size();x++)
        {
            if(arr[x]==0)ans[1]=x+1;
            if(arr[x]==2)ans[0]=x+1;
            if(ans[0]!=0 && ans[1]!=0)break;
        }
        return ans;
        
    }
};